types.ts

```diff
@@ -81,6 +81,13 @@ export enum AuthenticationType {
    * @see {@link OAuth2Authentication}
    */
   OAuth2 = 'OAuth2',
+  /**
+   * Authenticate using OAuth2 client credentials.
+   * The API must use a (largely) standards-compliant implementation of OAuth2.
```

Comment:
I would just change this line to be a link to https://oauth.net/2/grant-types/client-credentials/

This comment existed because for regular OAuth there are all kinds of APIs that _almost_ implement but do horrible quirky stuff. Some of it we support, some of it we don't. I don't think we've seen an implementation of client credentials that isn't compliant yet, which makes sense because it's a much simpler flow.